File: Typescript/slidingWindow/subarrayProductLessThanK.ts
var solutio = function(nums: number[], k: number): number {
    if (k <= 1) return 0
    let prod: number = 1
    let left: number = 0
    let count: number = 0

    for (let right = 0; right < nums.length; right++){
        prod *= nums[right]

        while (prod >= k) {
            prod = Math.floor(prod/nums[left])
            left ++
        }

        count += right - left + 1
    }

    return count
}

File: Python/Medium/sliding_window/subarrayProductLessThank.py
def solution(nums, k):
    if k <= 1:
        return 0
    
    prod = 1
    n = len(nums)
    left = 0
    count = 0


    for i in range(n):
        prod *= nums[i]

        while prod >= k:
            prod //= nums[left]
            left += 1
        
        count += i - left + 1

    return count
    